### ISSUE TITLE
'json/reader.h' file not found

### ISSUE THREAD

**Issue by davivid** [labatrockwell/ofxLibwebsockets#60]
Trying to compile with of_v0.8.4_osx_release I get the error above from file jsoncpp.cpp. What is the solution for this?

Thanks,
David

**Comment by davivid** [comment]
The release 0.8.4 + Bug fixes works as expected.

**Comment by robotconscience** [comment]
Hey David,
Let me look into this! Both addons include the same file, so you may just need to remove the include/libs/json folder from the ofxLibwebsockets project

**Comment by davivid** [comment]
Removing the reference to the include/libs/json folder seems to have solved it.
Cheers!

**Comment by robotconscience** [comment]
Great! I'll add a note to the readme.

**Comment by smallfly** [comment]
Hi,

I'm having similar issues, and I would like both add-ons (ofxLibwebsockets & ofxJSON) to be using their own version of JsonCpp. For example, the version of JsonCpp included in ofxLibwebsockets' repo has been modified.

Would it make sense to change the name of the ```jsoncpp.cpp``` parent directory, and to change the symbols used as include guards, so that they are unique to ofxLibwebsockets?

Cheers.

**Comment by aferriss** [comment]
I'm getting the same problem. It seems like the project structure has changed since @davivid 's advice above? I tried deleting the libs/jsoncpp folder and removing the header search paths but then ofxlibwebsockets complains about not being able to find jsoncpp at all. Any ideas?

**Comment by robotconscience** [comment]
@aferriss are you using ofxJson? and what version of OF?
We're going to really soon switch to ofJson once it's in the downloadable release (might already, actually), so this issue will be 100% fixed soon. Happy to help in the mean time!

**Comment by aferriss** [comment]
@robotconscience yes ofxJson and ofxLibwebsockets in tandem was my attempt, with 0.9.2. I ended up doing my parsing on the server side before having it sent to oF so I got it sorted! Seems like it will just be easiest to wait for 0.10

**Comment by visualist** [comment]
Hi, I too just ran into this with a 0.9.7 project with both ofxJSON and ofxLibwebsockets. I also see this #include lands within a !defined(JSON_IS_AMALGATED) section. Any recommendations on how I should build this?

**Comment by mattfelsen** [comment]
Hi @visualist. I haven't tried with 0.9.7 yet, but with older 0.9.x releases you should be able to remove the libs/jsoncpp files from one of the addons (I'm not sure if it matters whether you remove them from ofxJSON or ofxLibwebsockets) in your project to get around these errors.
